pkg/cloudinfo/alibaba/api_response.go

```diff
@@ -0,0 +1,72 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package alibaba
+
+import (
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/responses"
+	"github.com/banzaicloud/cloudinfo/pkg/cloudinfo"
+	"github.com/pkg/errors"
+	"strconv"
+	"time"
+)
+
+// CommonDescriber interface for all Alibaba API calls
+type CommonDescriber interface {
+	ProcessCommonRequest(request *requests.CommonRequest) (*responses.CommonResponse, error)
+}
+
+func (a *AlibabaInfoer) apiResponse(name, region string, instanceTypes []string) (*responses.CommonResponse, error) {
```

Comment:
split this switch - case statement and create individual factory-like methods for requests: NewDescribeHistoryRequest